demos/coreMQTT/mqtt_demo_keep_alive.c

```diff
@@ -0,0 +1,1177 @@
+                        MQTT_Status_strerror( xMQTTStatus ) ) );
+            xDemoStatus = pdFAIL;
+        }
+
+        /****************************** Connect. ******************************/
+
+        if( xDemoStatus == pdPASS )
+        {
+            /* Attempt to connect to the MQTT broker. If connection fails, retry
+             * after a timeout. The timeout value will be exponentially increased
+             * until the maximum number of attempts are reached or the maximum
+             * timeout value is reached. The function below returns a failure status
+             * if the TCP connection cannot be established to the broker after
+             * the configured number of attempts. */
+            xNetworkStatus = prvConnectToServerWithBackoffRetries( &xNetworkContext );
+
+            if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+            {
+                /* Errors are logged in prvConnectToServerWithBackoffRetries. */
+                xDemoStatus = pdFAIL;
+            }
+            else
+            {
+                /* Set a flag indicating a TCP connection exists. This is done to
+                 * disconnect if the loop exits before disconnection happens. */
+                xIsConnectionEstablished = pdTRUE;
+            }
+        }
+
+        if( xDemoStatus == pdPASS )
+        {
+            /* Sends an MQTT Connect packet over the already connected TCP
+             * socket, and waits for connection acknowledgment (CONNACK) packet. */
+            LogInfo( ( "Creating an MQTT connection to %s.",
+                       democonfigMQTT_BROKER_ENDPOINT ) );
+            xDemoStatus = prvCreateMQTTConnectionWithBroker( &xMQTTContext,
+                                                             &xNetworkContext );
+        }
+
+        if( xDemoStatus == pdPASS )
+        {
+            /* Create an auto-reload timer to handle keep-alive. */
+            xKeepAliveTimer = xTimerCreateStatic( "KeepAliveTimer",
+                                                  mqttexampleKEEP_ALIVE_DELAY,
+                                                  pdTRUE,
+                                                  ( void * ) &xMQTTContext.transportInterface,
+                                                  prvKeepAliveTimerCallback,
+                                                  &xKeepAliveTimerBuffer );
+
+            if( xKeepAliveTimer == NULL )
+            {
+                LogError( ( "xTimerCreateStatic() failed to create KeepAliveTimer." ) );
+                xDemoStatus = pdFAIL;
+            }
+        }
+
+        if( xDemoStatus == pdPASS )
+        {
+            /* Start the timer for keep-alive. */
+            xTimerStatus = xTimerStart( xKeepAliveTimer, 0 );
+
+            if( xTimerStatus != pdPASS )
+            {
+                LogError( ( "xTimerStart() failed to start KeepAliveTimer." ) );
+                xDemoStatus = pdFAIL;
```

Comment:
Can the `xTimerStatus` variable be avoided to instead just use `xDemoStatus`?
```suggestion
            xDemoStatus = xTimerStart( xKeepAliveTimer, 0 );

            if( xDemoStatus != pdPASS )
            {
                LogError( ( "xTimerStart() failed to start KeepAliveTimer." ) );
```

Reply:
Sure